lib/fog/libvirt/requests/compute/list_domains.rb

```diff
@@ -46,6 +46,24 @@ def boot_order xml
           xml_elements(xml, "domain/os/boot", "dev")
         end
 
+        def firmware(xml)
+          firmware_from_loader = xml_elements(xml, "domain/os/loader", "type").first
```

Comment:
Untested, but should this be
```suggestion
          firmware_from_loader = xml_element(xml, "domain/os/loader", "type")
```